feat(fclasses): Design Keys

wodenx opened the pull request:
## Changes
- Add "data-bl-design-key" attribute to designable elements.
- Control presence of this attribute based on "withShowDesighKeys" hoc
- Enable/disable in gatsby-theme-bodiless

## Related Issues

Closes #660

hvanyo reviewed (commented):
- Search: ?:SearchWrapper, ?:SearchInput, ?:SearchButton
- Iframe: on page: /iframe/ uses Embed:Wrapper, can we change it to use Iframe?
- Filters: /products under Filter:TagList there is unidentified list ?:Item
- RTE: /rte doesn't produce any bl-design-keys
- Image: /image doesn't produce any bl-design-keys as its not designable..  leave as is?
- Editable: are not producing bl-design-keys as its not designable..  leave as is?
- Links: are not producing bl-design-keys as its not designable..  leave as is?

wodenx commented:
> * Search: ?:SearchWrapper, ?:SearchInput, ?:SearchButton
> * Iframe: on page: /iframe/ uses Embed:Wrapper, can we change it to use Iframe?
> * Filters: /products under Filter:TagList there is unidentified list ?:Item
I'll fix these.  Search wasn't merged yet when I did this

> * RTE: /rte doesn't produce any bl-design-keys
I think we should leave this as is.  Not sure it will be common to apply design to components after they have been put into RTE.

> * Image: /image doesn't produce any bl-design-keys as its not designable..  leave as is?
> * Editable: are not producing bl-design-keys as its not designable..  leave as is?
> * Links: are not producing bl-design-keys as its not designable..  leave as is?
What w9uld you expect for these?  If they are not designable, how could they have design keys?

-

hvanyo reviewed (commented):
in .env.site can you this to serve as documentation (both test & starter site)
```
# Enable to help debug in static mode
# BODILESS_DEBUG='1'
```

wodenx commented:
> Search: ?:SearchWrapper, ?:SearchInput, ?:SearchButton

Fixed

> Iframe: on page: /iframe/ uses Embed:Wrapper, can we change it to use Iframe?

This is because it is the "Embed" component which is designable.  It can't be overridden, and it would be confusing to do so I think.

> Filters: /products under Filter:TagList there is unidentified list ?:Item

This is because the product filters are still using the deprecated list api. Should be fixed by #621 

> This case isn't working.
> set .env.site BODILESS_DEBUG='1', run npm build and serve -- verify bl-design-key's are in static mode

Fixed and added example to .env.site in test-site and starter.
